### PR TITLE
Add image member methods to openstackv2

### PR DESCRIPTION
**Pull request by vdloo**
## Add list image member methods to openstack v2 driver

includes and depends on https://github.com/apache/libcloud/pull/1151

this allows listing image members, see https://developer.openstack.org/api-ref/image/v2/index.html#sharing. in openstack you can share images between tenants. when an image is set to 'shared' other tenants can be added as 'member'. all members of an image can be listed to see which tenants have access and if they have accepted or rejected the membership yet or if it is still pending. this PR adds the new NodeImageMember class which contains image id, member id, state and more.

list image members

```
In [6]: conn.ex_list_image_members(image.id)
Out[6]: 
[<NodeImageMember: id=016926dff12345e8b10329f24c99745b, image_id=d9a9cd9a-278a-444c-90a6-d24b8c688a63, state=accepted, driver=OpenStack  ...>,
...

In [7]: image_member = conn.ex_list_image_members(image.id)[0]

In [8]: dir(image_member)
Out[8]: 
['__class__',
 '__delattr__',
 '__dict__',
 '__doc__',
 '__format__',
 '__getattribute__',
 '__hash__',
 '__init__',
 '__module__',
 '__new__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__setattr__',
 '__sizeof__',
 '__str__',
 '__subclasshook__',
 '__weakref__',
 '_uuid',
 'created',
 'driver',
 'extra',
 'get_uuid',
 'id',
 'image_id',
 'state',
 'uuid']

In [9]: image_member.extra
Out[9]: 
{'created': u'2018-01-12T12:31:50Z',
 'schema': u'/v2/schemas/member',
 'updated': u'2018-01-12T12:31:54Z'}
```

list image member:
```
In [7]: conn.ex_list_image_member(image.id, member.id)
Out[7]: <NodeImageMember: id=016926dff12345e8b10329f24c99745b, image_id=d9a9cd9a-278a-444c-90a6-d24b8c688a63, state=accepted, driver=OpenStack  ...>
```

### Status

- done, ready for review

### CODE REVIEW

**Comment by pquentin** [comment]
Can you please rebase your changes on top of the latest trunk? It would allow to see coverage info, since the latest trunk fixes an issue with sphinx. Thanks!

**Comment by vdloo** [comment]
yes no problem. also please see https://github.com/apache/libcloud/pull/1151, this PR depends on and includes that one

File: libcloud/compute/drivers/openstack.py [hunk 1]
```diff
@@ -2589,6 +2603,43 @@ def ex_update_image(self, image_id, data):
...
+            '/v2/images/%s/members' % (image_id,)
+        )
+        image_members = []
+        for image_member in response.object['members']:
+            image_members.append(self._to_image_member(image_member))
+        return image_members
+
+    def ex_list_image_member(self, image_id, member_id):
+        """
+        List a member of an image by id
```
Comment (pquentin) [inline libcloud/compute/drivers/openstack.py]: Is there a good reason to call this operation list and not get?

Comment (vdloo) [reply libcloud/compute/drivers/openstack.py]: Oops, has been changed to ex_get_image_member now to conform to the convention

**Comment by vdloo** [comment]
rebased on origin/trunk

**Comment by pquentin** [comment]
Thanks! Merged in trunk. ✨

**Pull request closed without merging**